### ISSUE TITLE
Support Thunderbird 68.0

### ISSUE THREAD

**Issue by lordi** [kewisch/quickmove-extension#3]
This is an awesome plugin I use daily.

It would be nice if the recent TB version would be supported.

**Comment by LispEngineer** [comment]
https://developer.thunderbird.net/add-ons/updates

These are some developer instructions on how to do it.

**Comment by kewisch** [comment]
You might want to try out https://github.com/kewisch/quickmove-extension/releases/tag/v2.0.0pre1 but please note this is work in progress!

**Comment by paescuj** [comment]
Thank you for your effort! :smiley: 
Unfortunately, it doesn't seem to work yet. The add-on fails to start:
```
[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIFile.initWithPath]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: chrome://quicktext/content/modules/wzQuicktext.jsm :: loadSettings :: line 249"  data: no] wzQuicktext.jsm:249:18
    loadSettings chrome://quicktext/content/modules/wzQuicktext.jsm:249
    <anonymous> chrome://quicktext/content/main.js:11
    loadScript resource:///modules/Overlays.jsm:561
    load resource:///modules/Overlays.jsm:226
    load resource:///modules/Overlays.jsm:48
    registerNonBootstrapped chrome://messenger/content/parent/ext-legacy.js:242
    forEach self-hosted:266
    registerNonBootstrapped chrome://messenger/content/parent/ext-legacy.js:236
    InterpretGeneratorResume self-hosted:1284
    AsyncFunctionNext self-hosted:839
```
Using Thunderbird 68.1.1 (64-bit) on Linux.

**Comment by boegel** [comment]
@kewisch Is there anything we can do to help, other than testing?

I'm totally clueless about implementing Thunderbird extensions to help out myself, but maybe posting an overview of known issues to fix would be useful?

**Comment by kewisch** [comment]
@paescuj this seems like an error related to the QuickText add-on?

@boegel testing is most of it. The add-on code is probably mostly good, but testers seem to be uncovering some bugs in the Thunderbird APIs I am using.

If something doesn't work when testing:
* Go to Tools > Developer Tools > Debug Add-ons 
* Click the `inspect` button for Quick Folder Move
* Note any messages in the "console" tab
* If there is an exception and you are good with debugging, check which API call threw the exception and note the parameters passed in.
* Also make sure you use the latest 68 point release.

**Comment by boegel** [comment]
@kewisch That's helpful, thanks, I'll see what I can come up with.

Any pointers on how to install an add-on using a source tarball (i.e. this repo)?

**Comment by kewisch** [comment]
If you just git clone it somewhere, you can go to the "debug add-ons" pane (called about:debugging), install a temporary add-on and then navigate to the manifest.json in the repository.

Once you've done this you can make changes and when you want to try them out just click the reload button in about:debugging.

**Comment by kewisch** [comment]
Thanks so much for digging into this!

**Comment by paescuj** [comment]
@kewisch Jup, seems like. Sorry about that! I'm going to take a second look tomorrow.

**Comment by paescuj** [comment]
@kewisch Sorry for the confusion! Now it's working with Thunderbird 68.2.2 on Linux :smiley: 
However, I'm not sure where I can find information about the new shortcuts, so I'm using the legacy shortcuts for now.

**Comment by lemmy04** [comment]
using the 1.7.0RC, keyboard shortcuts do not work.

Thunderbird 68.2.2, openSUSE Leap 15.1, KDE/Plasma desktop, shift-M is not "taken" by something else...

**Comment by kumarpranay** [comment]
MacOS 10.14.6
Was happpily using TB 60.0 with v1.6.1. Upgraded to TB 68.3.0 and am unable to use any of the pre versions. Both 2.0.0pre & 1.7.0pre do not show any menu items and none of the shortcuts work.

Tried to Debug the addon in the console and for both of the versions, I get the following error as soon as I click the "Preference" button for the addon.

> NS_ERROR_FILE_NOT_FOUND: Component returned failure code: 0x80520012 (NS_ERROR_FILE_NOT_FOUND) [nsIStyleSheetService.preloadSheet] ExtensionCommon.jsm:2554

The page opens and I am able to select the number of messages to keep in history. On the v2.0.0pre1 I can also see a button to enable legacy key bindings. When I press it shows a message and button disables briefly before going to back the initial state (no msg & button enabled)

**Comment by helmo** [comment]
See #9 , v1.7.0pre2 has this resolved.

**Comment by boegel** [comment]
Confirmed, v1.7.0pre2 works for me with Thunderbird 68.3.0 on macOS 10.15.2 :+1:

**Comment by kewisch** [comment]
Thanks for confirming, happy it works!

**Comment by kewisch** [comment]
Actually keeping this open until release

**Comment by helmo** [comment]
Please don't be afraid of bugs ... there will always be more to find. Anyone who updated Thunderbird is now unable to use this great plug-in unless they install it manually.

Publishing this to the regular add-ons site will allow more people to use it and provide feedback.

Thanks for working on this.

**Comment by kewisch** [comment]
Released 1.7.0 which works quite well. Thanks for bearing with me!
